### PR TITLE
fix start failed in enhanced kernel-native mode

### LINKED ISSUE #1099: kernel-native mode in kernel-enhanced cannot be build
**What happened**:
when i make build in kernel-enhanced linux, some vars are not be defined
```
   BUILD   kmesh-daemon
# kmesh.net/kmesh/pkg/bpf/ads
pkg/bpf/ads/sock_ops.go:98:11: sc.KmeshTailCallProg undefined (type *BpfSockOps has no field or method KmeshTailCallProg)
pkg/bpf/ads/sock_ops.go:106:11: sc.KmeshTailCallProg undefined (type *BpfSockOps has no field or method KmeshTailCallProg)
pkg/bpf/ads/sock_ops.go:124:11: sc.KmeshTailCallProg undefined (type *BpfSockOps has no field or method KmeshTailCallProg)
pkg/bpf/ads/sock_ops.go:142:11: sc.KmeshTailCallProg undefined (type *BpfSockOps has no field or method KmeshTailCallProg)
pkg/bpf/ads/sock_ops.go:211:31: sc.KmeshSockopsMaps.KmeshTailCallProg undefined (type enhanced.KmeshSockopsMaps has no field or method KmeshTailCallProg)
pkg/bpf/ads/loader_enhanced.go:95:21: sc.SockConn.KmeshConfigMap undefined (type BpfSockConn has no field or method KmeshConfigMap)
pkg/bpf/ads/loader_enhanced.go:117:62: sc.SockOps.KmeshSockopsMaps.KmeshListener undefined (type enhanced.KmeshSockopsMaps has no field or method KmeshListener)
pkg/bpf/ads/loader_enhanced.go:121:45: sc.SockOps.MapOfRouterConfig undefined (type BpfSockOps has no field or method MapOfRouterConfig)
pkg/bpf/ads/loader_enhanced.go:125:45: sc.SockOps.KmeshCluster undefined (type BpfSockOps has no field or method KmeshCluster)
pkg/bpf/ads/loader_enhanced.go:178:37: sc.SockOps.KmeshSockopsMaps.KmeshClusterStats undefined (type enhanced.KmeshSockopsMaps has no field or method KmeshClusterStats)
pkg/bpf/ads/loader_enhanced.go:178:37: too many errors
make: *** [Makefile:111: all-binary] Error 1

```
**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Kmesh version:
- Others:

### PR DESCRIPTION
**Pull request by lec-bit**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
1. start failed in enhanced kernel-native mode
2. fix workload tail_call wrong pin in SetMapPinType, move it in SockConnWorkload attach 

**Which issue(s) this PR fixes**:
Fixes #1099

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: pkg/bpf/utils/pin.go [hunk 1]
```diff
@@ -74,8 +74,8 @@ func UnpinMaps(value *reflect.Value) error {
 func SetMapPinType(spec *ebpf.CollectionSpec, pinType ebpf.PinType) {
 	for key, v := range spec.Maps {
 		// tail_call map dont support pinning when shared by different bpf types
-		if strings.HasPrefix(key, ".rodata") || key == ".bss" || key == "kmesh_tail_call_prog" ||
-			key == "kmesh_tail_call_ctx" {
+		if strings.HasPrefix(key, ".rodata") || key == ".bss" || key == "km_tailcallprog" ||
+			key == "km_tailcallctx" {
```
Comment (hzxuzhonghu) [inline pkg/bpf/utils/pin.go:78]: can we define some constants to remind developers to update it when they rename the map?

File: pkg/bpf/workload/sock_connection.go [hunk 2]
```diff
@@ -138,6 +138,10 @@ func (sc *SockConnWorkload) Attach() error {
 		Program: sc.KmeshCgroupSockWorkloadObjects.CgroupConnect6Prog,
 	}
 
+	// pin bpf_tail_call map
+	// tail_call map cannot pin in SetMapPinType->LoadAndAssign, we pin them independently
```
Comment (hzxuzhonghu) [inline pkg/bpf/workload/sock_connection.go:142]: >tail_call map cannot pin in SetMapPinType->LoadAndAssign

Why cannot be pinned? In dual-engine, the new xdp tailcall lead to this?

Comment (hzxuzhonghu) [reply pkg/bpf/workload/sock_connection.go:142]: Can you add more comments in the code

File: bpf/kmesh/ads/include/ctx/sock_ops.h [hunk 3]
```diff
@@ -10,6 +10,10 @@ typedef struct bpf_sock_ops ctx_buff_t;
 
 #define KMESH_PORG_CALLS sockops
 
+// tail_call map dont support pinning when shared by different bpf types, so define different name in sockops & sockconn
+#define map_of_tail_call_prog kmtailcallprogl7
+#define map_of_tail_call_ctx  kmtailcallctx_l7
```
Comment (hzxuzhonghu) [inline bpf/kmesh/ads/include/ctx/sock_ops.h:15]: in enhanced mode, the macro will conflict with that defined above in bpf/kmesh/ads/include/ctx/sock_addr.h

Comment (nlgwcy) [inline bpf/kmesh/ads/include/ctx/sock_ops.h:15]: No need modify.

File: bpf/kmesh/ads/include/ctx/sock_addr.h [hunk 4]
```diff
@@ -10,6 +10,10 @@ typedef struct bpf_sock_addr ctx_buff_t;
 #define KMESH_PORG_CALLS    cgroup/connect4
 // clang-format on
 
+// tail_call map dont support pinning when shared by different bpf types, so define different name in sockops & sockconn
+#define map_of_tail_call_prog km_tailcallprog
```
Comment (nlgwcy) [inline bpf/kmesh/ads/include/ctx/sock_addr.h:14]:
```suggestion
#define map_of_tail_call_prog km_cgrptailcall
```

Comment (lec-bit) [reply bpf/kmesh/ads/include/ctx/sock_addr.h:14]: ok

File: bpf/kmesh/ads/include/ctx/sock_ops.h (hunk 3 above)
Comment (lec-bit) [reply bpf/kmesh/ads/include/ctx/sock_ops.h:15]: no, skops use sock_ops.h , and sockconn use sock_addr.h
so will not conflict

File: bpf/kmesh/ads/include/ctx/sock_addr.h [hunk 5]
```diff
@@ -10,6 +10,9 @@ typedef struct bpf_sock_addr ctx_buff_t;
 #define KMESH_PORG_CALLS    cgroup/connect4
 // clang-format on
 
+// tail_call map dont support pinning when shared by different bpf types, so define different name in sockops & sockconn
+#define map_of_tail_call_prog km_cgrptailcall
```
Comment (hzxuzhonghu) [inline bpf/kmesh/ads/include/ctx/sock_addr.h:14]: For other reviewers, this is making the map named `km_cgrptailcall` in sock_addr prog

And below make anothe map `km_skopstailcall` in sock_ops prog. 

So the code can be reused `kmesh_tail_call` by different progs without passing in the map name

**Pull request merged**